### PR TITLE
tpmmanager: fix build, switch to Qt5

### PR DESCRIPTION
**Pull request by wamserma**
###### Description of changes

- fixed hash
- switched deps from Qt4 to Qt5

ZHF: #172160

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/tpmmanager/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, qt4, qmake4Hook, trousers }:
+{ lib, stdenv, fetchFromGitHub, fetchpatch, qt5, wrapQtAppsHook, trousers }:
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tpmmanager/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, fetchpatch, qtbase, qmake, wrapQtAppsHook, trousers }:
```

File: pkgs/applications/misc/tpmmanager/default.nix [hunk 2]
```diff
@@ -8,12 +8,20 @@ stdenv.mkDerivation rec {
...
+      # build with Qt5
+      url = "https://patch-diff.githubusercontent.com/raw/Rohde-Schwarz/TPMManager/pull/6.patch";
+      sha256 = "sha256-iNuMn8S+vqjE5xsdxbA1fIs/2ApLLEsJw4XYfWOmSGA=";
+    })
+  ];
 
-  buildInputs = [ qt4 trousers ];
+  nativeBuildInputs = [ qt5.qmake wrapQtAppsHook ];
+
+  buildInputs = [ qt5.qtbase trousers ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tpmmanager/default.nix:24]:
```suggestion
  nativeBuildInputs = [ qmake wrapQtAppsHook ];

  buildInputs = [ qtbase trousers ];
```

File: pkgs/applications/misc/tpmmanager/default.nix [hunk 3]
```diff
@@ -8,12 +8,20 @@ stdenv.mkDerivation rec {
...
     rev = "v${version}";
-    sha256 = "sha256-JKYG+I/tZ+0NDmHcIgKV6eGrjbPvPQKPo0sE/zBlLY4=";
+    sha256 = "sha256-UZYn4ssbvLpdB0DssT7MXqQZCu1KkLf/Bsb45Rvgm+E=";
   };
 
-  nativeBuildInputs = [ qmake4Hook ];
+  patches = [
+    (fetchpatch {
+      # build with Qt5
+      url = "https://patch-diff.githubusercontent.com/raw/Rohde-Schwarz/TPMManager/pull/6.patch";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/tpmmanager/default.nix:17]: Please fetch https://github.com/Rohde-Schwarz/TPMManager/commit/f62c0f2de2097af9b504c80d6193818e6e4ca84f.patch and https://github.com/Rohde-Schwarz/TPMManager/commit/c287a841ac6b057ed35799949211866b9f533561.patch instead.

File: pkgs/applications/misc/tpmmanager/default.nix [hunk 4]
```diff
@@ -8,12 +8,24 @@ stdenv.mkDerivation rec {
     owner = "Rohde-Schwarz";
     repo = "TPMManager";
     rev = "v${version}";
-    sha256 = "sha256-JKYG+I/tZ+0NDmHcIgKV6eGrjbPvPQKPo0sE/zBlLY4=";
+    sha256 = "sha256-UZYn4ssbvLpdB0DssT7MXqQZCu1KkLf/Bsb45Rvgm+E=";
```
Comment (ajs124) [inline pkgs/applications/misc/tpmmanager/default.nix:11]: why did the hash change, if the rev didn't?

Comment (wamserma) [reply pkgs/applications/misc/tpmmanager/default.nix:11]: The old hash was probably wrong. The build failed and the hash needed updating to fix this.
The hash was last touched when this was converted to ffGH: https://github.com/NixOS/nixpkgs/commit/3a1e015627474c8f1da6a5c83577cd3e293a7475

ping @Stunkymonkey

Comment (Stunkymonkey) [reply pkgs/applications/misc/tpmmanager/default.nix:11]: I can confirm, that the new hash is correct. No idea where I got mine from.

**Review by Artturin — APPROVED** [review]
builds and i was able to see the error mesage about not having tscd or something

**Pull request merged**